package test;

import static org.junit.Assert.*;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

class NoteControleDaoTest {

    private NoteControleDao dao;
    
    @BeforeEach
    void setUp() throws Exception {
        dao = new NoteControleDao();
    }

    @Test
    void testGetByControle() {
        try {
            ArrayList<NoteControle> result = dao.getByControle(25);
            assertNotNull(result);
            assertTrue(result.size() > 0);
            for(NoteControle noteControle: result) {
                assertNotNull(noteControle.getId());
                assertNotNull(noteControle.getEtudiant());
                assertNotNull(noteControle.getNote());
                assertNotNull(noteControle.getValide());
                assertNotNull(noteControle.getObservation());
            }
        } catch (SQLException e) {
            fail("Exception thrown: " + e.getMessage());
        }
    }
}
